python/cudf/cudf/core/dataframe.py

```diff
@@ -6454,12 +6473,45 @@ def _apply_support_method(self, method, axis=0, *args, **kwargs):
                 "support `bool_only`."
                 raise NotImplementedError(msg)
 
-            prepared = self._prepare_for_rowwise_op()
+            prepared = self._prepare_for_rowwise_op(method)
+            if not skipna and any(col.nullable for col in prepared._columns):
+                mask = cudf.DataFrame(
+                    {
+                        name: prepared._data[name]._get_mask_as_column()
+                        if prepared._data[name].nullable
+                        else column.full(len(prepared._data[name]), True)
+                        for name in prepared._data.names
+                    }
+                )
+                mask = mask.all(axis=1)
+            else:
+                mask = None
+
+            for col in prepared._data.names:
+                if prepared._data[col].nullable:
+                    prepared._data[col] = (
+                        prepared._data[col]
+                        .astype(
+                            cudf.utils.dtypes.get_min_float_dtype(
+                                prepared._data[col]
+                            )
+                        )
+                        .fillna(np.nan)
+                    )
```

Comment:
In the case where I have something like one `int64 non-null` and one `int64 nullable` column, this will make `result` eventually be a `float64` type. Any chance we could detect this condition and typecast back to `int64` for the result?

Reply:
To perform that we'll probably have to then check for if all dtypes are homogeneous and depending on operation decide if we want to type-cast the result back or not. i.e., In (`mean`, `std`, ..) we should not type-cast and in (`min`, `max`, ...) we can type-cast back.

But I think type-casting might not give us intended lower memory as in `as_gpu_matrix` we are changing to a common dtype(which will be entire matrix of `float32`/`float64` even if one column of dataframe has a null value).

Reply:
It's not about lower memory usage, it's about expectations for type consistency.

I guess what I'm thinking is that before this block of code we do something like: `result_type = find_common_type(prepared.dtypes)` then after this block we do `result = result.astype(result_type, copy=False)`

Reply:
Changed the logic to handle the expected output dtypes(special cased to specific methods)